assets/section-main-product.css

```diff
@@ -533,9 +533,13 @@ a.product__text {
 }
 
 @media screen and (max-width: 749px) {
-  .product__media-item--variant:first-child {
+  .product__media-item--variant:nth-last-child(2):is(.product__media-item--variant) {
```

Comment:
And I was just thinking. If we apply the fix for the spacing on sliders for Safari, then that would fix this issue as well. 
And I think it might make sense to apply it to all sliders 🤔 
The fix is adding a pseudo element to the `ul` to add the spacing at the end of the slider. https://github.com/Shopify/dawn/pull/1648

Reply:
I made some changes to fix all sliders actually so they get the whitespace at the end in Safari as well as it's fixing this specific issue as well: https://github.com/Shopify/dawn/issues/1668

I haven't test a whole a lot but it seems to work from what I could tell. 

The number of slides in the counter seem to be off when we're using big values for the grid horizontal spacing but I can replicate them on main as well 🤔 (different scenarios tho)  So I don't think it's introduced in here.